src/query/storages/fuse/src/io/locations.rs

```diff
@@ -39,32 +39,52 @@ static SNAPSHOT_STATISTICS_V0: TableSnapshotStatisticsVersion =
 #[derive(Clone)]
 pub struct TableMetaLocationGenerator {
     prefix: String,
+    part_prefix: String,
 }
 
 impl TableMetaLocationGenerator {
     pub fn with_prefix(prefix: String) -> Self {
-        Self { prefix }
+        Self {
+            prefix,
+            part_prefix: "".to_string(),
+        }
+    }
+
+    pub fn with_part_prefix(mut self, part_prefix: String) -> Self {
+        self.part_prefix = part_prefix;
+        self
     }
 
     pub fn prefix(&self) -> &str {
         &self.prefix
     }
 
+    pub fn part_prefix(&self) -> &str {
+        &self.part_prefix
+    }
+
     pub fn gen_block_location(&self) -> (Location, Uuid) {
         let part_uuid = Uuid::new_v4();
-        (
-            (
-                format!(
-                    "{}/{}/{}_v{}.parquet",
-                    &self.prefix,
-                    FUSE_TBL_BLOCK_PREFIX,
-                    part_uuid.as_simple(),
-                    DataBlock::VERSION,
-                ),
+        let location_path = if self.part_prefix.is_empty() {
+            format!(
+                "{}/{}/{}_v{}.parquet",
+                &self.prefix,
+                FUSE_TBL_BLOCK_PREFIX,
+                part_uuid.as_simple(),
                 DataBlock::VERSION,
-            ),
-            part_uuid,
-        )
+            )
+        } else {
+            format!(
+                "{}/{}/{}{}_v{}.parquet",
+                &self.prefix,
+                FUSE_TBL_BLOCK_PREFIX,
+                &self.part_prefix,
+                part_uuid.as_simple(),
+                DataBlock::VERSION,
+            )
+        };
+
+        ((location_path, DataBlock::VERSION), part_uuid)
```

Comment:
It looks the if and else blocks can be merged into one block.

Reply:
Yup